src/neo/Network/P2P/Helper.cs

```diff
@@ -6,15 +7,30 @@ namespace Neo.Network.P2P
     public static class Helper
     {
         public static byte[] GetHashData(this IVerifiable verifiable)
+        {
+            return GetHashData(verifiable, ProtocolSettings.Default.Magic);
+        }
+
+        public static byte[] GetHashData(this IVerifiable verifiable, uint magic)
         {
             using (MemoryStream ms = new MemoryStream())
             using (BinaryWriter writer = new BinaryWriter(ms))
             {
-                writer.Write(ProtocolSettings.Default.Magic);
+                writer.Write(magic);
                 verifiable.SerializeUnsigned(writer);
                 writer.Flush();
                 return ms.ToArray();
             }
         }
+
+        public static UInt256 CalculateHash(this IVerifiable verifiable)
+        {
+            return CalculateHash(verifiable, ProtocolSettings.Default.Magic);
```

Comment:
I prefer to avoid multiple calls or please add `AgressiveInline`

```suggestion
            return new UInt256(Crypto.Hash256(verifiable.GetHashData(ProtocolSettings.Default.Magic)));
```